receiver/dotnetdiagnosticsreceiver/config.go

```diff
@@ -0,0 +1,45 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnetdiagnosticsreceiver
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+type Config struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	// Optional directory name to save binary blobs read from the socket. This can
+	// be useful for troubleshooting/debugging. If left blank or unset, no blobs are
+	// saved.
+	BlobDir string `mapstructure:"blob_dir"`
+	// The maximum number of blob files to keep in the blob_dir, if specified. Defaults
+	// to 10 (most recent).
+	MaxBlobFiles int `mapstructure:"max_blob_files"`
+	// The process ID of the dotnet process from which to collect diagnostics. This
+	// receiver is intended to be used with an observer and receiver creator for
+	// process discovery.
+	PID int `mapstructure:"pid"`
+	// The duration between collection of metrics. The duration value is converted
+	// to seconds and sent to the dotnet backend at receiver startup time.
+	// Defaults to 1 second.
+	CollectionInterval time.Duration `mapstructure:"collection_interval"`
```

Comment:
Consider to use the scraper helper if possible

Reply:
The way this receiver works is maybe a little unusual. We send a request over a Unix domain socket, and part of that request has the collection interval in seconds that we want. The dotnet process then sends us metrics at that interval, and when they are received, they are parsed and sent down the collector pipeline. So the timing is managed by the dotnet process after that initial request.

Reply:
You can use the settings from the scraper at least correct?

Reply:
Sure -- latest change uses scraper settings. Thanks.